File: 프로그래머스/lv2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
from collections import deque
def solution(queue1, queue2):
    q1 = deque(queue1)
    q2 = deque(queue2)
    if (sum(q1) + sum(q2))%2 == 1:
        return -1
    q1_sum = sum(q1)
    q2_sum = sum(q2)
    cnt = 0
    while True:
        if q1_sum > q2_sum:
            a = q1.popleft()
            q2.append(a)
            q1_sum-=a
            q2_sum+=a
            cnt+=1
        elif q1_sum < q2_sum:
            b = q2.popleft()
            q1.append(b)
            q1_sum+=b
            q2_sum-=b
            cnt+=1
        else:
            break
        if cnt>300001:
            return -1
    return cnt

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.js
function solution(queue1, queue2) {
    let cnt=0
    let q1=0
    let q2=0
    let idx1=0
    let idx2=0
    for(let n of queue1) q1+=n
    for(let n of queue2) q2+=n
    if((q1+q2)%2===1) return -1
    while(true){
        if(q1>q2){
            q1-=queue1[idx1]
            q2+=queue1[idx1]
            queue2.push(queue1[idx1])
            idx1++
            cnt++
        } else if(q1<q2){
            q2-=queue2[idx2]
            q1+=queue2[idx2]
            queue1.push(queue2[idx2])
            idx2++
            cnt++
        } else{
            return cnt
        } 
        if(idx1>300001 || idx2>300001) return -1
    }
}